pkgs/desktops/gnome-3/extensions/gsconnect/default.nix

```diff
@@ -0,0 +1,77 @@
+{ stdenv, fetchFromGitHub, substituteAll, python3, openssl, which
+, meson, ninja, libxml2, pkgconfig, gobjectIntrospection, wrapGAppsHook
+, glib, gtk3, folks, nautilus, upower, sshfs, gnome3 }:
+
+stdenv.mkDerivation rec {
+  name = "gnome-shell-gsconnect-${version}";
+  version = "20180922.1";
+
+  src = fetchFromGitHub {
+    owner = "andyholmes";
+    repo = "gnome-shell-extension-gsconnect";
+    rev = "e4be2356815cb9863f0c02776dae3c7235e2f3a5";
+    sha256 = "070i8cyc4xi44j7pcdggpnhppchysmdj6i59vcnpy27rgkpgdwx6";
+  };
+
+  patches = [
+    (substituteAll {
+      src = ./fix-paths.patch;
+      inherit sshfs openssl which;
+      python3 = python3.withPackages [ python3.pkgs.pygobject3 ];
```

Comment:
We will need to rely on shebang and running the [script](https://github.com/andyholmes/gnome-shell-extension-gsconnect/blob/e4be2356815cb9863f0c02776dae3c7235e2f3a5/src/service/components/folks.py) directly, instead of through [`python3`](https://github.com/andyholmes/gnome-shell-extension-gsconnect/blob/e4be2356815cb9863f0c02776dae3c7235e2f3a5/src/service/components/contacts.js#L285) to be able to use `wrapGAppsHook` on it. The [direct library use](https://github.com/andyholmes/gnome-shell-extension-gsconnect/blob/e4be2356815cb9863f0c02776dae3c7235e2f3a5/src/service/components/folks.py#L18) is also concerning.

Reply:
This line fails on my system:
```
error: attempt to call something which is not a function but a list, at /home/etu/code/nixpkgs/pkgs/development/interpreters/python/with-packages.nix:3:19
(use '--show-trace' to show detailed location information)
```

Reply:
Should be `python3.withPackages (pkgs: [ python3.pkgs.pygobject3 ])` but it is still broken until we fix it as described above.